JASP-Engine/JASP/R/binomialtestbayesian.R

```diff
@@ -15,497 +15,239 @@
+  
+  # Take results from state if possible
+  if (!is.null(jaspResults[["stateBayesianBinomResults"]])) return(jaspResults[["stateBayesianBinomResults"]]$object)
+  
+  # This will be the object that we fill with results
+  results <- list()
+  
+  # First, we perform precalculation of variables we use throughout the analysis
+  results[["spec"]] <- .binomCalcSpecs(dataset, options)
+  results[["binom"]] <- list()
+  
+  for (variable in options$variables) {
+    
+    results[["binom"]][[variable]] <- list()
+    
+    # Prepare for running the binomial test
+    column <- dataset[[.v(variable)]]
+    data   <- column[!is.na(column)]
+    levels <- levels(data)
+    
+    for (level in levels) {
+      n   <- length(data)
+      counts <- sum(data == level)
+      prop   <- counts / n
+      
+      # Binomial test for plots
+      plotResults <- stats::binom.test(
+        x           = counts,
+        n           = n,
+        p           = options$testValue,
+        alternative = "two.sided",
+        conf.level  = options$descriptivesPlotsConfidenceInterval
+      )
+      
+      # Summary statistics for plots
+      plotDat <- data.frame(
+        label      = level,
+        proportion = prop,
+        lowerCI    = plotResults$conf.int[1],
+        upperCI    = plotResults$conf.int[2]
+      )
+      
+      # beta distribution parameters
+      a <- options$priorA
+      b <- options$priorB
+      
+      if (options$hypothesis == "notEqualToTestValue") 
+        hyp <- "two.sided"
+      else if (options$hypothesis == "greaterThanTestValue")
+        hyp <- "greater"
+      else
+        hyp <- "less"
+      
+      BF10  <- .bayesBinomialTest(counts, n, options$testValue, hypothesis = hyp, a = a, b = b)
+      
+      BF <- BF10
+      
+      if (options$bayesFactorType == "BF01") {
+        BF <- 1/BF10
+      } else if(options$bayesFactorType == "LogBF10") {
+        BF <- log(BF10)
+      }
+      
+      # Add results for each level of each variable to results object
+      results[["binom"]][[variable]][[level]] <- list(
+        case          = variable,
+        level         = level,
+        counts        = counts,
+        total         = n,
+        proportion    = prop,
+        BF            = BF,
+        plotDat       = plotDat
+      )
+    }
+  }
+  
+  # Save results to state
+  jaspResults[["stateBayesianBinomResults"]] <- createJaspState(results)
+  jaspResults[["stateBayesianBinomResults"]]$dependOn(
+    c("variables", "testValue", "hypothesis", "descriptivesPlotsConfidenceInterval")
+  )
+  
+  # Return results object
+  return(results)
+}
+
+# Main Table ----
+.bayesianBinomTableMain <- function(jaspResults, dataset, options, bayesianBinomResults, errors){
+  if (!is.null(jaspResults[["bayesianBinomialTable"]])) return()
+  variables <- unlist(options$variables)
+  theta0<- .5
+  
+  bayesianBinomialTable                   <- createJaspTable("Bayesian Binomial Test")
```

Comment:
Please fix the spacing here and elsewhere. In general, there should be a single space before and after binary operators (`+`, `-`, `==`, etc.) and assignment operators (`<-`). Exceptions are for aligning multiple lines.